test/test_sympy_utils.py

```diff
@@ -520,6 +523,74 @@ def test_z3_proof_floordiv_eq_simplify(self):
         r = solver.check()
         self.assertEqual(r, z3.unsat)
 
+class TestSingletonInt(TestCase):
+    def test_basic(self):
+        j1 = SingletonInt(1, 1)
+        j1_copy = SingletonInt(1, 1)
+        j2 = SingletonInt(2, 1)
+        j1x2 = SingletonInt(1, 2)
+
+        # eq
+        self.assertTrue(sympy.Eq(j1, j1))
+        self.assertTrue(sympy.Eq(j1, j1_copy))
+        self.assertFalse(sympy.Eq(j1, j2))
+        self.assertFalse(sympy.Eq(j1, j1x2))
+        self.assertFalse(sympy.Eq(j1, sympy.Integer(1)))
+        self.assertFalse(sympy.Eq(j1, sympy.Integer(3)))
+
+        # ne
+        self.assertFalse(sympy.Ne(j1, j1))
```

Comment:
Instead of testing this manually, make a helper function that tests for Eq, and then tests Ne is the inverse of Eq

Reply:
Ditto for the relationals below